finder-app/writer.c

```diff
@@ -0,0 +1,52 @@
+/* Assignment -2 Intro to System Programming
+ * Author : Swapnil Ghonge
+ * Details: This files contains fucntions which coorelate with writer.sh				
+*/
+#include<stdio.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include<string.h>
+#include<unistd.h>
+#include<syslog.h>
+
+
+int main(int argc, char*argv[])
+{
+	openlog(NULL, 0, LOG_USER);
+	
+	//Error Checking for input of number of arguements
+	if (argc !=3)
+	{
+		syslog(LOG_ERR,"Wrong number of arguemnts your input arguments are %d", argc);
+		
+		return 1;
+		
+	}
+	//Error checking for null string
+	if (*argv[2] =='\0')
+	{
+		syslog(LOG_ERR, "Invalid String");
+	}
+
+	else
+	{
+		syslog(LOG_DEBUG,"File created successfully: %s",argv[1]);
```

Comment:
@swapnilghonge for readability you should consider using a named variable rather than using argv directly throughout.